core/src/test/scala/io/finch/BodySpec.scala

```diff
@@ -11,14 +12,20 @@ class BodySpec extends FinchSpec {
 
   behavior of "body*"
 
-  def withBody(b: String): Input = Input.post("/").withBody[Text.Plain](Buf.Utf8(b))
+  def withBody[A : Show](b: A): Input = Input.post("/").withBody[Text.Plain](Buf.Utf8(Show[A].show(b)))
 
-  checkAll("Body[Int]", EntityEndpointLaws[Int](stringBodyOption)(withBody).evaluating)
-  checkAll("Body[Long]", EntityEndpointLaws[Long](stringBodyOption)(withBody).evaluating)
-  checkAll("Body[Boolean]", EntityEndpointLaws[Boolean](stringBodyOption)(withBody).evaluating)
-  checkAll("Body[Float]", EntityEndpointLaws[Float](stringBodyOption)(withBody).evaluating)
-  checkAll("Body[Double]", EntityEndpointLaws[Double](stringBodyOption)(withBody).evaluating)
-  checkAll("Body[UUID]", EntityEndpointLaws[UUID](stringBodyOption)(withBody).evaluating)
+  implicit def decodeBodyEntity[A](implicit de: DecodeEntity[A]): Decode.Text[A] = Decode.text[A]((b, _) => {
+    val Buf.Utf8(str) = b
+    de(str)
+  })
+
+  checkAll("Body[Int]", EntityEndpointLaws[Int](textBodyOption)(withBody).evaluating)
```

Comment:
So I think `textBodyOption` doesn't have to follow the laws from `EntityEndpointLaws`. Maybe `stringBodyOption` does (I'm not sure)? I'd probably vote for just removing these checks for `strinbBodyOption` for now.